### ISSUE TITLE
autconf + boostrap.sh broken?

### ISSUE THREAD

**Issue by pierrejoye** [libgd/libgd#746]
➜  libgd_actions git:(features/mingw-ci) ✗ ./bootstrap.sh
/bin/sh: 0: Illegal option --

running manually the part I need:

 ```
autoreconf -f -i

 is m4_require'd but not m4_defun'dr_inttypes_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
 is m4_require'd but not m4_defun'dr_stdint_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
 is m4_require'd but not m4_defun'dr_strings_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
 is m4_require'd but not m4_defun'dr_unistd_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
 is m4_require'd but not m4_defun'dr_sys_stat_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
 is m4_require'd but not m4_defun'dr_sys_types_h
../../lib/autoconf/headers.m4:273: AC_CHECK_HEADERS_ONCE is expanded from...
configure.ac:64: the top level
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: copying file 'config/ltmain.sh'
.ibtoolize:   error: AC_CONFIG_MACRO_DIRS([m4]) conflicts with ACLOCAL_AMFLAGS=-I m4
autoreconf: libtoolize failed with exit status: 1
```

which auto* version do we use? autoconf 2.69 here.

After having regenerated it manuallu (automake and co):

checking that generated files are newer than configure... done

configure: creating ./config.status
.in'ig.status: error: cannot find input file: `Makefile

**Comment by cmb69** [comment]
Are there CRLF in bootstrap.sh?

**Comment by pierrejoye** [comment]
all LF afaics

did you try it lately?

**Comment by cmb69** [comment]
Nope, didn't try it, but I had similar issues in the past due to checking out bash scripts on Windows. Running something like
````
git config --system core.autocrlf false
git config --system core.eol lf
````
before the checkout might solve it.

**Comment by pierrejoye** [comment]
What I did, let me try again

**Comment by pierrejoye** [comment]
```
➜  libgd cd libgd_autoconf
➜  libgd_autoconf git:(master) ./bootstrap.sh
/bin/sh: 0: Illegal option --
➜  libgd_autoconf git:(master) sh ./bootstrap.sh
: not found.sh: 3:
./bootstrap.sh: 28: Syntax error: end of file unexpected (expecting "then")
➜  libgd_autoconf git:(master)
```

same

**Comment by pierrejoye** [comment]
```
core.autocrlf=false
core.eol=lf
user.email=pierre.php@gmail.com
user.name=Pierre Joye
core.autocrlf=true
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
remote.origin.url=git@github.com:libgd/libgd.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.master.remote=origin
branch.master.merge=refs/heads/master
```

**Comment by pierrejoye** [comment]
I did not expect to have to go through autoconf mess again lol

If only we could support only cmake but apparently configure is still used by a lot. @remicollet or?

**Comment by pierrejoye** [comment]
ok found the culpitr

```
➜  libgd_autoconf git:(master) find . -name \*.m4|xargs dos2unix
dos2unix: converting file ./m4/ax_require_defined.m4 to Unix format...
dos2unix: converting file ./m4/ax_append_flag.m4 to Unix format...
dos2unix: converting file ./m4/w32.m4 to Unix format...
dos2unix: converting file ./m4/visibility.m4 to Unix format...
dos2unix: converting file ./m4/ax_pthread.m4 to Unix format...
dos2unix: converting file ./m4/ax_cflags_warn_all.m4 to Unix format...
dos2unix: converting file ./m4/ax_openmp.m4 to Unix format...
```

Somehow I can't manage to commit them. fresh checkout or not, with autocrlf to false.

**Comment by pierrejoye** [comment]
thanks @cmb69 :)

I added a .gitattribute now, should not happen anymore :)
